okhttp/src/main/kotlin/okhttp3/internal/socket/OkioServerSocketFactory.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2024 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package okhttp3.internal.socket
+
+import javax.net.ServerSocketFactory
+
+interface OkioServerSocketFactory {
+  fun newServerSocket(): OkioServerSocket
+}
+
+class RealOkioServerSocketFactory(
```

Comment:
> Also the name OkioServerSocketFactory is awkward once you later drop Okio, should we think about the name already?

I don't have strong feelings about this, but I think the `Okio*` names might be ideal specifically because it disambiguates them from the stdlib equivalents. I can't think of any other name I'd pick. That said, maybe a better name will reveal itself as we flesh them out?

Reply:
Yep, works for me.